virtcontainers/container.go

```diff
@@ -304,6 +305,32 @@ func (c *Container) mountSharedDirMounts(hostSharedDir, guestSharedDir string) (
 			continue
 		}
 
+		var stat unix.Stat_t
+		if err := unix.Stat(m.Source, &stat); err != nil {
+			return nil, err
+		}
+
+		// Check if mount is a block device file. If it is, the block device will be attached to the host
+		// instead of passing this as a shared mount.
+		if c.checkBlockDeviceSupport() && stat.Mode&unix.S_IFBLK == unix.S_IFBLK {
+			b := &BlockDevice{
+				DeviceType: DeviceBlock,
+				DeviceInfo: DeviceInfo{
+					HostPath:      m.Source,
+					ContainerPath: m.Destination,
+					DevType:       "b",
+				},
+			}
+
+			// Attach this block device, all other devices passed in the config have been attached at this point
+			if err := b.attach(c.pod.hypervisor, c); err != nil {
```

Comment:
This is being attached but where are we detaching (unplugging) this device from the VM ? We need to make sure the device is properly detached when the container is stopped(destroyed).

Reply:
@sboeuf  This has been taken care of now.

Reply:
Do you need to check for duplication? Or is it OK to attach/detach the same device multiple times?

Reply:
@bergwolf I had not seen any issues in attaching/detaching same device files multiple times to a VM. However I tried mounting the device files at two different locations and am seeing some sync issues. I think it would be necessary to keep track at a pod level if the device has already been passed and not pass it again. But I am thinking of opening a separate PR to handle that issue, as this is an existing issue for block devices passed with --device as well. What do you think @bergwolf ?

Reply:
@amshinde I'm not sure that I followed you. AFAIK, when we mount the same block device at different mount points, they share the same superblock data structure in the kernel, which makes sure data is always consistent. What sync issues did you see with that kind of setup?

Reply:
@bergwolf Right now if a device file say "/dev/sdb" is attached twice, I did not see any issues in attaching it twice and detaching it. However, since the file is attached twice, it may appear twice inside the VM. (say as "/dev/vda" and "/dev/vdb"). Sync issues will arise if you mount those two.

Reply:
@amshinde @bergwolf I have performed the same testing last week and I can confirm that if you pass a device/volume to a container A, and you modify the content of the mounted filesystem by adding a new file for instance, a new container B created after this will see the change. But unfortunately, after both container are running, any change is only reported back to the host but not sync'ed back to the guest.